### DISCUSSION TITLE
Creating template repositories via CLI/API

### DISCUSSION THREAD

**Discussion by jgbm** [community/community#29312] [Other Feature Feedback, Questions, & Ideas]
I'm attempting to automate the creation of repositories to serve as GitHub Classroom assignments.  The only current fly in the ointment is that I can't seem to set a repository as a template repository via the `gh` command line tool.  Is there some setting I'm missing?  If not, is there a way to do it via the API?

**Comment by mldyh** [comment]
hi @jgbm,  Based on this documentation, https://docs.github.com/en/rest/repos/repos#create-an-organization-repository you can create new repository and set it as template by send `is_template=true` in you body request. Hope it helps.. 🥂
